core/src/main/java/org/elasticsearch/index/query/QueryParseContext.java

```diff
@@ -321,7 +321,7 @@ public QueryBuilder parseInnerFilterToQueryBuilder(String queryName) throws IOEx
     @Deprecated
     public Query parseInnerFilter(String queryName) throws IOException, QueryParsingException {
         QueryBuilder builder = parseInnerFilterToQueryBuilder(queryName);
-        return (builder != null) ? builder.toQuery(this) : null;
+        return (builder != null) ? builder.toQuery(new QueryGenerationContext(this)) : null;
```

Comment:
all these above methods should be gone once we are done with all queries I think? then it is fine to create a new object all the time, this is all temporary. Maybe mark //norelease these methods?